src/main/kotlin/org/rust/lang/core/resolve/ref/RsMarcoItemReferenceContributor.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.resolve.ref
+
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.*
+import com.intellij.util.ProcessingContext
+import org.rust.lang.core.psi.RsElementTypes.IDENTIFIER
+
+class RsMarcoItemReferenceContributor : PsiReferenceContributor() {
+    override fun registerReferenceProviders(registrar: PsiReferenceRegistrar) {
+        val pattern = PlatformPatterns.psiElement(IDENTIFIER)//never called
```

Comment:
I think completion should be handled by `CompletionProvider`. Doing completion via the usual reference machinery won't work, because we don't know if we are typing a macro until we see `!`.